lib/discovery.js

```diff
@@ -41,12 +45,12 @@ function appendToChain(c: Chain, addresses: Array<string>): Chain {
     if (addresses.length === 0) {
         return c;
     }
-    let ca = new Map(c.addresses);
-    let ci = new Map(c.indexes);
+    let ca = c.addresses;
+    let ci = c.indexes;
     addresses.forEach((a, i) => {
         i += c.nextIndex;
-        ca.set(i, a);
-        ci.set(a, i);
+        ca = ca.set(i, a);
+        ci = ci.set(a, i);
```

Comment:
huh, I was under the impression that creations of new maps are cheap in immutable, but it seems they are not. ok, will change